### PR TITLE
[ICU-680] Fix Duplicate posts

### PR DESCRIPTION
**Pull request by enahum**
#### Summary
Removed the race condition that made the message being posted twice cause of a false/possitive and now when a new post is created we will add the pending post id to the list until the post is actually created.

In case we do not have a response from the server while the post is in a pending state and the app is sent to the background the pending post is cleared from the store and if in fact the post was created on the server it will be received with the next get posts request.

#### Ticket Link
https://mattermost.atlassian.net/browse/ICU-680

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [ ] All new/modified APIs include changes to [mattermost-redux](https://github.com/mattermost/mattermost-redux) (https://github.com/mattermost/mattermost-redux/pull/401)

### CODE REVIEW

File: app/actions/views/channel.js [hunk 1]
```diff
@@ -149,13 +149,15 @@ export function loadPostsIfNecessaryWithRetry(channelId) {
 
         const time = Date.now();
 
-        let loadMorePostsVisible;
+        let loadMorePostsVisible = true;
```
Comment (hmhealey) [inline app/actions/views/channel.js]: Are these changes related to this PR?

Comment (enahum) [reply app/actions/views/channel.js]: nope but it was throwing errors if there was no connection to the server

**Comment by enahum** [comment]
this needed a mattermost-redux upgrade, will do in another PR
